README.md

```diff
@@ -231,6 +280,38 @@ lspconfig.SERVER.setup{config}
     `new_config.cmd = cmd` is present within the function body.
 ```
 
-# Individual server documentation
+## Debugging
+The two most common points of failure are 
 
-See [CONFIG.md](CONFIG.md) for documentation and configuration of individual language servers.
+1. The language server is not installed. You should be able to run the `cmd` 
+defined in each lua module from the command line.
+
+2. Not triggering root detection. The language server will only start if it
+is opened in a directory, or child directory, containing a file which signals
+the *root* of the project. Most of the time, this is a `.git` folder, but each server
+defines the root config in the lua file.
+
+Before reporting a bug, check your logs. Add the following to your init.vim
+to enable logging 
+
+```vim
+lua << EOF
+vim.lsp.set_log_level("debug")
+EOF
+```
+Attempt to run the language server, and view the log present at:
+```
+:lua print(vim.lsp.get_log_path())
```

Comment:
I can't find the magic invocation to open that file directly using the `get_log_path` function, but if anyone knows that, this might be better :)

Reply:
`v:lua.vim.lsp.get_log_path()` returns the log path, but I don't know how to open the returned string rather than a buffer of the function name itself 😆

Reply:
I got it: `:lua vim.cmd('e'..vim.lsp.get_log_path())`

Reply:
Should we open in a split by default? vsp?

Reply:
I'd keep it simple...

(Alternatively, you could spell it out: `~/.local/share/nvim/lsp.log` under Linux/BSD/macOS, `~\AppData\Local\nvim-data\lsp.log` under Windows?)

Reply:
We could wait until https://github.com/neovim/neovim/pull/13707 is merged I guess

Reply:
Went ahead with`:lua vim.cmd('e'..vim.lsp.get_log_path())`